service/javadsl/client/src/main/java/com/lightbend/lagom/internal/javadsl/CircuitBreakersPanelImpl.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.javadsl;
+
+import akka.actor.ActorSystem;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.lightbend.lagom.internal.client.CircuitBreakerConfig;
+import com.lightbend.lagom.internal.client.CircuitBreakersPanelInternal;
+import com.lightbend.lagom.internal.spi.CircuitBreakerMetricsProvider;
+import com.lightbend.lagom.javadsl.client.CircuitBreakersPanel;
+import scala.compat.java8.FutureConverters;
+import scala.concurrent.Future;
+import scala.runtime.AbstractFunction0;
+
+import java.util.concurrent.CompletionStage;
+import java.util.function.Supplier;
+
+@Singleton
+public class CircuitBreakersPanelImpl implements CircuitBreakersPanel {
+
+  private final CircuitBreakersPanelInternal circuitBreakersPanelInternal;
+
+  @Inject()
+  public CircuitBreakersPanelImpl(ActorSystem system,
+                                  CircuitBreakerConfig config,
+                                  CircuitBreakerMetricsProvider metricsProvider) {
+
+    circuitBreakersPanelInternal = new CircuitBreakersPanelInternal(system, config, metricsProvider);
+  }
+
+  @Override
+  public <T> CompletionStage<T> withCircuitBreaker(String id, Supplier<CompletionStage<T>> body) {
+
+    Future<T> tFuture = circuitBreakersPanelInternal.withCircuitBreaker(
+            id,
+            new AbstractFunction0<Future<T>>() {
+              @Override
+              public Future<T> apply() {
+                return FutureConverters.toScala(body.get());
```

Comment:
This at least compiles:

```java
  @Override
  public <T> CompletionStage<T> withCircuitBreaker(String id, Supplier<CompletionStage<T>> body) {
        return FutureConverters.toJava(
                circuitBreakersPanelInternal.withCircuitBreaker(id,
                        (JFunction0<Future<T>>) () -> FutureConverters.toScala(body.get()))
        );
  }
```

Unfortunately, the `JFunction0` cast is necessary to hint the compiler, but it's totally safe.